internal/objects/contour/contour.go

```diff
@@ -106,7 +106,9 @@ func OtherContoursExistInSpecNs(ctx context.Context, cli client.Client, contour
 	}
 	if exist {
 		for _, c := range contours.Items {
-			if c.Spec.Namespace.Name == contour.Spec.Namespace.Name {
+			if c.Name != contour.Name &&
+				c.Namespace != contour.Namespace &&
+				c.Spec.Namespace.Name == contour.Spec.Namespace.Name {
```

Comment:
should this be `if (c.Name != contour.Name || c.Namespace != contour.Namespace) && c.Spec.Namespace.Name == contour.Spec.Namespace.Name {`? If it has a different namespace *or* name than `contour`, then it's not the same one.

Reply:
@skriss `OtherContoursExist()` will return a list of contours, including the one that's provided as input to `OtherContoursExistInSpecNs()`. L109-L110 ensures the Contour from the list is not the same as the input Contour. I could update `OtherContoursExist()` to filter the input Contour from the list so it only returns Contours that differ from the input Contour. Thoughts?

Reply:
I think it should be an `||` instead of an `&&`, since the one from the list needs to differ either by name, *or* by namespace (or both) in order to not be the same as the input Contour.

Reply:
```go
func different(c1, c2 *operatorv1alpha1.Contour) bool {
    return c1.Name != c2.Name || c1.Namespace != c2.Namespace
}
```

Reply:
@skriss thanks for catching this. I simplified the conditional logic in commit `38fd172`, PTAL when you have a moment.